packages/dx-react-demos/src/demo-viewer/demo-viewer.jsx

```diff
@@ -1,37 +1,71 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { Route, Switch } from 'react-router-dom';
 
 import { demos } from '../demo-registry';
 import { ThemeViewer } from './theme-viewer';
 import { DemoRenderer } from './demo-renderer';
 
+const SourceCode = ({ theme, section, demo }) => (
+  <div className="clearfix">
+    <a
+      className="pull-right"
+      style={{
+        marginTop: `${theme === 'bootstrap3' ? -10 : 10}px`,
+      }}
+      href={`https://github.com/DevExpress/devextreme-reactive/tree/master/packages/dx-react-demos/src/${theme}/${section}/${demo}.jsx`}
+    >
+      Source Code
+    </a>
+  </div>
+);
+
+SourceCode.propTypes = {
+  section: PropTypes.string.isRequired,
+  demo: PropTypes.string.isRequired,
+  theme: PropTypes.string.isRequired,
+};
+
 export const DemoViewer = ({
-  match: { params: { demo: currentDemo, section: currentSection } },
+  match: { params: { demo: currentDemo, section: currentSection }, url },
 }) => (
-  <ThemeViewer
-    avaliableThemes={Object.keys(demos[currentSection][currentDemo])}
-  >
-    {({ theme }) => (
-      <div>
-        <DemoRenderer
-          theme={theme}
-          section={currentSection}
-          demo={currentDemo}
-        />
-        <div className="clearfix">
-          <a
-            className="pull-right"
-            style={{
-              marginTop: `${theme === 'bootstrap3' ? -10 : 10}px`,
-            }}
-            href={`https://github.com/DevExpress/devextreme-reactive/tree/master/packages/dx-react-demos/src/${theme}/${currentSection}/${currentDemo}.jsx`}
-          >
-            Source Code
-          </a>
+  <Switch>
+    <Route
+      path={`${url}/clear/:theme`}
```

Comment:
Use 'clean' instead of 'clear'.
Try to use the route of the following format: `${url}/:theme/clean`.